### DISCUSSION TITLE
How to build it to cipher verison

### DISCUSSION THREAD

**Discussion by nineskys** [gilzoide/unity-sqlite-net#34] [Ideas]
Most of the time, a cryptographic system is required for a project

**Comment by gilzoide** [comment]
> How to build it to cipher verison

Hey @nineskys. There's an overview of what is needed [here](https://github.com/gilzoide/unity-sqlite-net/issues/28#issuecomment-2430296590).
I'm thinking on changing this package to build using SQLCipher instead of vanilla SQLite by default, but I can't work on this for the next couple of weeks, so it will take a while.

> Most of the time, a cryptographic system is required for a project

For save systems this is debatable, it depends a lot on the requirements of the project. Not all games require locking save data and it may just be ok for people to mess with the save and risk corrupting their game.
But SQLite is much more than just a tool for save systems, you can use databases as assets, or store data in-memory and quickly order/filter them, etc...

**Reply by gilzoide** [reply]
I know it's been a while, but I finally changed SQLite implementation to use [SQLite3 Multiple Ciphers](https://utelle.github.io/SQLite3MultipleCiphers/), so that databases can now be encrypted! I'll release a new version of this package with the new implementation, will let you know.
